test/inductor/test_torchinductor_opinfo.py

```diff
@@ -202,6 +202,7 @@ def process(device_type):
     "complex": {f16, f32, f64},
     "corrcoef": {f32, f64, i32, i64},
     "cov": {f32, f64, i32, i64},
+    "exponential": {f16, f32, f64},
```

Comment:
why these skips?

Reply:
The above didn't pass `TestInductorOpInfo` `test_comprehensive` (same for cauchy https://github.com/pytorch/pytorch/actions/runs/3999976098/jobs/6864738744), but I'm actually not sure why. 

I see there is https://github.com/pytorch/pytorch/blob/b399007a073d3a9ba4f6668c00d368fc9ee0f59a/test/inductor/test_torchinductor_opinfo.py#L253 which without gives the same error
`torch._dynamo.exc.Unsupported: call_function UserDefinedObjectVariable(uniform_) [TensorVariable(), ConstantVariable(int), ConstantVariable(int)] {}`

From the error log, it complains that  `VariableTracker.call_function` of `UserDefinedObjectVariable(uniform_)` is unimplemented, but I'm not sure why

Reply:
cc @wconstab, does this error look familiar? Did you fix some similar device/dtype related issue recently?

Reply:
no, this is something else.

that was related to 'default arguments' of functions, the fix added support if the functino had a default such as 
`def f(..., some_arg=torch.device('cuda'))`

the clue for this stacktrace is 
`  File "/opt/conda/envs/py_3.7/lib/python3.7/site-packages/torch/_dynamo/variables/user_defined.py", line 262, in call_function
`

I think probably we are failing one of these checks 

```
        if (
            self.is_supported_random()
            and all(k.is_python_constant() for k in args)
            and all(v.is_python_constant() for v in kwargs.values())
        ):
```

and then giving up and saying we can't `call_function` on this object

Reply:
Thanks @ngimel, @wconstab,

I had a quick test and even when the above checks would fail, it's able to `call_function` of `UserDefinedObjectVariable(exponential_)`
```
import torch

op = torch.Tensor.exponential_
args = (torch.empty(10), 0.5)

def foo():
    return op(*args)

opt_foo = torch.compile(foo)
y_ = opt_foo()
```

Reply:
So what's the difference between the above and failing test?

Reply:
Actually, setting `nopython=True` gives the same error. So as @wconstab hinted, the checks fail and graph breaks into errors throwing the msg `torch._dynamo.exc.Unsupported: call_function UserDefinedObjectVariable(exponential_) [TensorVariable(), ConstantVariable(float)] {}`. May I know why `test_torchinductor` tests for `nopython=True` ? 

```
import torch
import torch._dynamo as dynamo

op = torch.Tensor.exponential_
args = (torch.empty(10), 0.5)

def foo(args):
    return op(*args)

opt_foo = dynamo.optimize("inductor", nopython=True)(foo)
y_ = opt_foo(args) 
```

Reply:
Actually, why a torch method like `torch.Tensor.exponential_` would graph break? I had a quick try and while `torch.bernoulli` not graph break `torch.Tensor.bernoulli_` does

```
import torch
import torch._dynamo as dynamo

op = torch.bernoulli # no graph break
#op = torch.Tensor.bernoulli_ # graph break
args = torch.empty(10)

def foo(args):
    return op(args)

opt_foo = dynamo.optimize("inductor", nopython=True)(foo)
y_ = opt_foo(args)
```

Reply:
Setting `nopython=False` allows graph breaks, so the decompositions you are adding are pointless until the problem of graph breaks is solved.

Reply:
> until the problem of graph breaks is solved

Thanks @ngimel, I think the in-place sampling ops graph breaks are removable though (unlike necessary graph break from outside torch method, control flow, etc)